### PR TITLE
feat: [CM-632][Sale Widget] Add configuration flat to enable/disable transaction finality confirmation

### PR DESCRIPTION
**Pull request by mimi-imtbl**
### Hi👋, please prefix this PR's title with:

- [ ] `breaking-change:` if you have introduced modification that necessitates immediate adjustments by this SDK's users to their applications, clients, or integrations to avert disruptions to existing features or functionalities.
- [x] `feat:`, `fix:`, `refactor:`, `docs:`, or `chore:`.

## Added 

Added `waitFulfillmentSettlements` to sale widget config to enable/disable wait for transaction settlement. `true` by default.

### CODE REVIEW

**Review by jwhardwick — DISMISSED** [review]

File: packages/checkout/widgets-lib/src/widgets/sale/SaleWidgetRoot.tsx [hunk 1]
```diff
@@ -132,6 +138,9 @@ export class Sale extends Base<WidgetType.SALE> {
                   excludePaymentTypes={this.parameters.excludePaymentTypes!}
                   language="en"
                   multicurrency={!!this.properties?.config?.multicurrency}
+                  waitFulfillmentSettlements={
```
Comment (jhesgodi) [inline packages/checkout/widgets-lib/src/widgets/sale/SaleWidgetRoot.tsx:141]: Question and suggestion?
I think we may want the wait to be enabled by default, right? If so then I suggest you use
```
waitFulfillmentSettlements={this.properties?.config?.waitFulfillmentSettlements ?? true}
```
this expresion will allow`false`, but if null or undefined will default to `true`

File: packages/checkout/widgets-lib/src/widgets/sale/context/SaleContextProvider.tsx [hunk 2]
```diff
@@ -136,6 +136,7 @@ const SaleContext = createContext<SaleContextValues>({
   excludePaymentTypes: [],
   multicurrency: false,
   selectedCurrency: undefined,
+  waitFulfillmentSettlements: false,
```
Comment (jhesgodi) [inline packages/checkout/widgets-lib/src/widgets/sale/context/SaleContextProvider.tsx:139]: default to `true`

**Review by jhesgodi — CHANGES_REQUESTED** [review]
Looks good, left comment re initial state

File: packages/checkout/widgets-lib/src/widgets/sale/context/SaleContextProvider.tsx (hunk 2 above)
Comment (mimi-imtbl) [reply packages/checkout/widgets-lib/src/widgets/sale/context/SaleContextProvider.tsx:139]: good catch @J

**Review by jwhardwick — APPROVED** [review]

**Review by jhesgodi — APPROVED** [review]

**Pull request merged**